scenario/context.py

```diff
@@ -600,42 +589,29 @@ def _record_status(self, state: "State", is_app: bool):
         else:
             self.unit_status_history.append(state.unit_status)
 
-    def manager(self, event: "_Event", state: "State"):
+    def __call__(self, event: "_Event", state: "State"):
```

Comment:
does it make sense to return-type-annotate with `_Manager`?

Reply:
My experience is that pyright etc can generally infer return types so they aren't really needed unless you want it to be different than what would be inferred (Iterable instead of a list, for example). I'm pretty certain it will be inferring a `ManagedAction|ManagedEvent` return type here, so making `_Manager` explicit doesn't seem to gain much.

I guess the exception would be in documentation, either generated reference docs or in the pop-up type thing that IDEs do. I'm not sure that showing the return type helps much there, though? Especially if it's private.

Am I missing a reason to add it?

Reply:
I was expecting the union of action+event manager would be confusing enough to mess up the checker, but you're right. if pyright is smart enough to figure it out, we don't have to worry.